src/Resolver.js

```diff
@@ -217,19 +217,29 @@ class Resolver {
     return pkg;
   }
 
-  getPackageMain(pkg) {
+  async getPackageMain(pkg) {
     // libraries like d3.js specifies node.js specific files in the "main" which breaks the build
     // we use the "module" or "jsnext:main" field to get the full dependency tree if available
-    let main = [pkg.module, pkg['jsnext:main'], pkg.browser, pkg.main].find(
-      entry => typeof entry === 'string'
-    );
+    const entries = [pkg.module, pkg['jsnext:main'], pkg.browser, pkg.main]
+      .filter(entry => typeof entry === 'string')
+      .map(entry => path.resolve(pkg.pkgdir, entry));
+
+    // main is the first entry that exists
+    for (let i = 0; i < entries.length; ++i) {
+      const entry = entries[i];
 
-    // Default to index file if no main field find
-    if (!main || main === '.' || main === './') {
-      main = 'index';
+      try {
+        const stat = await fs.stat(entry);
```

Comment:
Maybe instead of doing the actual resolution here (and then duplicating it again later), we could just have `getPackageMain` return an array of possible paths. Then, `loadDirectory` could just loop through them and try each one: https://github.com/parcel-bundler/parcel/blob/master/src/Resolver.js#L188-L191